### PR TITLE
feat: Implemented new issue description [CFG-1574]

### PR DESCRIPTION
**Pull request by ofekatr**
- [x] Follows [CONTRIBUTING](https://github.com/snyk/snyk/blob/master/CONTRIBUTING.md) rules

#### What does this PR do?

- Implements the new issue description design for the Snyk IaC test output.
- Applies the new issue description design in the test flow, wrapped in the `iacCliOutput` feature flag.

#### Where should the reviewer start?

- `src/lib/formatters/iac-output/v2/issues-list/index.ts`

#### How should this be manually tested?

- Execute the `snyk iac test` command without the `iacCliOutput` feature flag, and ensure the output is the same as the original.
- Execute the `snyk iac test` command with the `iacCliOutput` feature flag, ensure the issues list is displayed, and each issue is displayed with the expected values and [design] (https://www.figma.com/file/buNFZqr1Uhf4SaifHw4WXE/CLI-IaC-Output-Improvements?node-id=0%3A1).

#### Any background context you want to provide?

- As part of the MVP for the new CLI output, we’d like to replace the current details section offered per issue, with a new, more descriptive one. The new details added to this section will allow users to get additional information about these issues, as well as guidance on how to resolve them.

#### What are the relevant tickets?

- [CFG-1574](https://snyksec.atlassian.net/browse/CFG-1574)

#### Screenshots

- Before
    image

- After
    image

### CODE REVIEW

**Comment by YairZ101** [comment]
I'm still in the middle of reviewing the PR but I will really appreciate if in the next time you will separate code refactoring from feature implementation.
This PR includes a lot of changes which are not necessarily relevant to the core feature implementation.

**Comment by YairZ101** [comment]
I'm still in the middle of reviewing the PR but I will really appreciate if in the next time you will separate code refactoring from feature implementation.
This PR includes a lot of changes which are not necessarily relevant to the core feature implementation.

**Review by YairZ101 — COMMENTED** [review]
Other than the small comments I wrote below it is looking great, awesome job @ofekatr!

File: src/lib/formatters/iac-output/v2/issues-list/issue.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    ['Rule', issue.documentation],
+    ['Path', printPath(issue.cloudConfigPath, 0)],
+    ['File', issue.targetFile],
+    [
+      'Resolve',
+      remediationKey && issue.remediation?.[remediationKey]
+        ? issue.remediation[remediationKey]
+        : issue.iacDescription.resolve,
+    ],
+  ].filter(([, val]) => !!val) as [string, string][];
```
[the remark this answers is not in the document]
Comment (ofekatr) [reply src/lib/formatters/iac-output/v2/issues-list/issue.ts:46]: There will be cases in which some of the issue properties may be missing.
For example, according to the `AnnotatedIssue` type, the `documentation` property could potentially be undefined. Additionally, for custom rules, there will be missing properties like the `documentation`, as well as some optional fields, based on how users define their rules.

Generally speaking though, it also serves as a safety measure to ensure we don't print any missing values as `undefined`.

**Comment by ofekatr** [comment]
>I will really appreciate it if in the next time you will separate code refactoring from feature implementation.
This PR includes a lot of changes which are not necessarily relevant to the core feature implementation.

Good call @YairZ101, I see what you mean. In retrospect, I should've made detected and separated the refactoring and the new implementation here before getting the two mixed. 
Will prioritize it more in future PRs.

File: src/lib/formatters/iac-output/v2/issues-list/types.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import { AnnotatedIacIssue } from '../../../../snyk-test/iac-test-result';
+import { IacOutputMeta } from '../../../../types';
+
+export const severityValues = [...Object.values(SEVERITY)] as const;
+
+export type SeverityTitleBySeverity = {
+  [severity in typeof severityValues[number]]: string;
+};
+
+export const SeverityTitleBySeverity = severityValues.reduce(
```
[the remark this answers is not in the document]
Comment (ofekatr) [reply src/lib/formatters/iac-output/v2/issues-list/types.ts]: Duly noted, it was changed according to your suggestion in [4640de6](https://github.com/snyk/cli/pull/3145/commits/4640de67f1076add8b3a6c0bce103615459b3117)

File: src/lib/formatters/iac-output/v2/issues-list/types.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  (severityTitles, severity: string) => {
+    severityTitles[severity] = capitalize(severity);
+    return severityTitles;
+  },
+  {} as SeverityTitleBySeverity,
+);
+
+export interface AnnotatedIssue extends AnnotatedIacIssue {
```
[the remark this answers is not in the document]
Comment (ofekatr) [reply src/lib/formatters/iac-output/v2/issues-list/types.ts:11]: - `PolicyMetadata` defines properties that are describe the security rule.
- `AnnotatedIacIssue` adds additional information which describe the issue generated by this security rule.

For the same properties described in the policy metadata, there can be multiple instances of an annotated IaC issue, which would add properties such as the 'Introduced By' path, which is defined per issue.

[the remark this answers is not in the document]
Comment (ofekatr) [reply src/lib/formatters/iac-output/v2/issues-list/types.ts]: I've updated the implementation to have a closer resemblance to the previous version. 
Before, the formatted object included the policy metadata fields in the `policyMetadata` property. 
After, the formatted object included the policy and issue details in the `issue` property.
The formatted object is considered a formatted result, and it currently includes the issue, target file, and project type.

File: src/lib/formatters/iac-output/v2/issues-list/formatters.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
+import { IacTestResponse } from '../../../../snyk-test/iac-test-result';
+import { IacOutputMeta } from '../../../../types';
+import { TestData } from './types';
+
+export function formatScanResultsNewOutput(
+  oldFormattedResults: IacTestResponse[],
```
[the remark this answers is not in the document]
Comment (ofekatr) [reply src/lib/formatters/iac-output/v2/issues-list/formatters.ts]: There are minor typing differences between the two as it seems they were created for different purposes, yet in some locations, they are used interchangeably.

I've updated the code to uniformly use the `FormattedResult` type.

**Pull request merged**